src/main/java/upb/dice/twitter/TweetExtractor.java

```diff
@@ -1,56 +1,117 @@
 package upb.dice.twitter;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import twitter4j.*;
 
+import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.util.Date;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 
 /**
+ * An implementation of the Data Extractor for twitter
  * If the rate limit is not reached, extract tweets based on the keywords or location (provided as latitude, longitude and the radius of coverage)
  * and stores them in a file. Also maintains a trackrecord in a file of the maxID obtained in every search
  */
-public class TweetExtractor {
-    private StoreMaxID storeMaxID = new StoreMaxID();
+public class TweetExtractor implements DataExtractor {
+
+    private final static String directoryName = "Tweets_Search_Details";
+    private final static String TweetDataFilepath = directoryName + File.separator + "Tweet_object1.txt";
+    public long oldestTweetID;
+    private Logger LOGGER = LoggerFactory.getLogger(TweetExtractor.class);
+    private IDHandler idHandler = new IDHandler();
     private Twitter twitter = TwitterFactory.getSingleton();
-    private RateLimitChecker rateLimitChecker = new RateLimitChecker();
-    private boolean check;
+    private boolean remainingTweet;
+    public boolean rateLimit;
+    private long maxId;
+    Query queryDup;
+    long sinceID;
+    int counter = 0;
 
     /**
      * Recursive search for tweets based on this query and store the maxID for the search results
```

Comment:
I am wondering why the word "Recursive" is used. The used algorithm is not recursive, is it? :thinking: